### DISCUSSION TITLE
Error installing Petsc

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#367] [Installation]
Hi. Everything seemed to be going fine with the installation (I use Ubuntu 22.04 on a VM virtual box running on windows). 

Then when installing Petsc I got this error ([error_petsc.txt](https://github.com/mdolab/dafoam/files/9754407/error_petsc.txt)). I also attach the configure.log file ([configure.log](https://github.com/mdolab/dafoam/files/9754386/configure.log))

What could be the issue? This percolated into my subsequent installations - I got an error while trying to install MACH-Aero.

I got the error in the first step in the screenshot (then configure and compile)
![image](https://user-images.githubusercontent.com/10186475/195073798-d8a06ae7-674b-4818-9909-2bdd28820e22.png)

because of which the next step (finally install...) gave me this error 

[error_petsc_2.txt](https://github.com/mdolab/dafoam/files/9754448/error_petsc_2.txt)

**Comment by friedenhe** [comment]
DAFoam is not tested on Ubuntu 22.04. Please use Ubuntu 20.04

**Reply by uttamcadambi07** [reply]
Thank you.
I removed Ubuntu 22.04 LTS and installed 20.04 LTS.

I compiled DAFoam on this version and everything went by fine (no errors in all the packages).

But when I tried to run the regression tests, I got the error $HOME/dafoam/repos/dafoam/tests: No such file or directory. Has the path to the directory containing the regression tests changed or did I miss something?

**Reply by uttamcadambi07** [reply]
PROBLEM RESOLVED.

Dr. He, please update the installation page and change the path to the tests from $HOME/dafoam/repos/dafoam/tests to $HOME/dafoam/repos/dafoam-3.0.2/tests for DAFoam version 3.0.2.

Thanks!

**Reply by friedenhe** [reply]
Thanks for the feedback. Docs updated.

**Reply by lucy2649** [reply]
Dear Professor Ping,
I am encountering the same issue when running the installation commands on Ubuntu 22.04.3. The problem persists even when building Docker images from scratch using FROM ubuntu:22.04.

The installation consistently gets stuck at the following step:

make PETSC_DIR=$DAFOAM_ROOT_PATH/packages/petsc-${PETSC_VER} PETSC_ARCH=real-opt all -j ${MAKE_JOBS} && \
    cd $PETSC_DIR/src/binding/petsc4py && pip install .

[installpetsc4pyerror.txt](https://github.com/user-attachments/files/23397213/installpetsc4pyerror.txt)

The process hangs at this stage and does not proceed further. Could you please provide any suggestions on how to resolve this issue?
Thank you for your assistance.

**Reply by friedenhe** [reply]
Can you follow the exact steps/commands to compile on Ubuntu 22.04? Also, are you using native Ubuntu or in a virtual machine? The petsc installation command you showed seems to be different from what we have on the DAFoam website.

**Reply by lucy2649** [reply]
I am experiencing PETSc compilation errors on native Ubuntu 24.04.3 LTS when attempting to install DAFoam using both methods:

When following the "Compile from source" instructions for Ubuntu, the PETSc compilation fails. Error log is attached: 
[CompilefromsourcePetscerror.txt](https://github.com/user-attachments/files/23407592/CompilefromsourcePetscerror.txt)

I created a Dockerfile with the installation commands and ran:
bashdocker build -t my-dafoam:v4.0.2-custom .
The Docker build also fails with the following error:
[BuildDockerimagesfromscratcherror.txt](https://github.com/user-attachments/files/23407673/BuildDockerimagesfromscratcherror.txt)

**Comment by friedenhe** [comment]
We have not tested DAFoam on Ubuntu 24.04 yet.... Please use Ubuntu 22.04 instead.

**Reply by lucy2649** [reply]
I reinstalled Ubuntu 22.04.2, but after running it, the following error still occurs: 
Compile from source error: 
[CompilefromsourceUbuntunstallpetsc4pyerror.txt](https://github.com/user-attachments/files/23452339/CompilefromsourceUbuntunstallpetsc4pyerror.txt)

The Docker build also fails with the following error:
[BuildDockerimagesfromscratcherror.txt](https://github.com/user-attachments/files/23452342/BuildDockerimagesfromscratcherror.txt)

**Reply by lucy2649** [reply]
Hi, I created a working version of the Dockerfile based on the official one.
[Dockerfile.txt](https://github.com/user-attachments/files/23468628/Dockerfile.txt)

The main changes involve PETSc + petsc4py, MACH-Aero components, and FUNTOFEM.

I tried to preserve the original functionality, but I’m not sure if there might be any minor differences.

**Comment by friedenhe** [comment]
Hmm. not sure why the petsc4py keeps giving you an error, but I am glad you made a new Dockerfile to make it work.

**Reply by lucy2649** [reply]
I suspect it's a command version compatibility issue.
Running "pip install ."causes it to automatically update to the latest version (instead of using the specified version), and subsequently results in a path naming mismatch error.

**Comment by friedenhe** [comment]
@lucy2649 We believe the error you saw may be related to a very subtle bug in the pip install . environment. pip install . is trying to use cython 3.x to build petsc4py, even cython 0.x is used in DAFoam. So the fix is to run `pip install . --no-build-isolation -v` to force the pip command to use the DAFoam's cython. Can you try? We have just updated the installation guide with the new command.

**Reply by lucy2649** [reply]
I am experiencing the same issue even after using the updated version. The error is attached.
[dockertest.txt](https://github.com/user-attachments/files/23769706/dockertest.txt)

**Comment by friedenhe** [comment]
@lucy2649 can you try again? We just updated the installation guide and the docker build guide

**Reply by lucy2649** [reply]
I then tried building the Docker image from scratch, and this time it worked! 
Thanks for the update.
